molo/core/tests/base.py

```diff
@@ -301,9 +301,19 @@ def mk_translation(self, source, language, translation):
         language_relation = translation.languages.first()
         language_relation.language = language
         language_relation.save()
+        translation.language = language
         translation.save_revision().publish()
+        source.specific.translated_pages.add(translation)
+        source.save()
         PageTranslation.objects.get_or_create(
             page=source, translated_page=translation)
+        for translated_page in \
+                source.specific.translated_pages.all():
+            translations = source.specific.translated_pages.all().\
+                exclude(language__pk=translated_page.language.pk)
+            for t in translations:
+                translated_page.translated_pages.add(t)
+            translated_page.save()
```

Comment:
Please add a comment to the above block. I'm struggling to understand what it is doing.